src/reducers/projects.js

```diff
@@ -83,6 +83,21 @@ export function reduceRoot(stateIn, action) {
             projects.get(currentProjectKey),
           );
         }
+      case 'FOCUS_LINE':
+        {
+          const currentProjectKey =
+            stateIn.getIn(['currentProject', 'projectKey']);
+
+          if (isNil(currentProjectKey)) {
+            return stateIn;
+          }
+
+          return projects.updateIn(
+            [currentProjectKey, 'hiddenUIComponents'],
+            hiddenUIComponents =>
+              hiddenUIComponents.remove(`editor.${action.payload.language}`),
+          );
+        }
```

Comment:
Definitely a good thought—I’m kind of on the fence and am planning a bigger refactor of the reducers very soon, so I’m comfortable with a bit of duplication for now.